Firebase/Messaging/FIRMessaging.m

```diff
@@ -177,12 +177,14 @@ + (FIRMessaging *)messaging {
   return (FIRMessaging *)messaging;
 }
 
-- (instancetype)initWithAnalytics:(nullable id<FIRAnalyticsInterop>)analytics {
+- (instancetype)initWithAnalytics:(nullable id<FIRAnalyticsInterop>)analytics
+                   withInstanceID:(FIRInstanceID *)instanceID
+                 withUserDefaults:(NSUserDefaults *)defaults {
   self = [super init];
   if (self != nil) {
     _loggedMessageIDs = [NSMutableSet set];
-    _instanceID = [FIRInstanceID instanceID];
-    _messagingUserDefaults = [NSUserDefaults standardUserDefaults];
+    _instanceID = instanceID ?: [FIRInstanceID instanceID];
```

Comment:
Ah, apologies I was thinking keeping them `nonnull` and below on line 215 and 216 changing the call to:

```
return [[FIRMessaging alloc] initWithAnalytics:analytics
                                                           instanceID:[FIRInstanceID instanceID]
                                                       userDefaults:[NSUserDefaults standardUserDefaults];
```

(Although this way may keep tests cleaner potentially, so it's up to you! If you'd prefer to keep this, please mark the two parameters with `nullable`.)